app/src/main/java/com/example/mytestproject/ui/weatherData/WeatherDataFragment.kt

```diff
@@ -39,12 +41,35 @@ class WeatherDataFragment : Fragment() {
         animationDrawable.start()
 
         setupWeatherData()
+
+        btnRetry.setOnClickListener {
+            weatherDataViewModel.getWeather()
+            setupWeatherData()
```

Comment:
There is no need to subscribe to LiveData every click on button. One time in onViewCreated is enough.